rtos/rtx/TARGET_CORTEX_M/cmsis_os.h

```diff
@@ -69,11 +69,15 @@
 #endif
 
 // The stack space occupied is mainly dependent on the underling C standard library
+#ifdef OS_MAINSTKSIZE
+#define WORDS_STACK_SIZE OS_MAINSTKSIZE
+#else
```

Comment:
Any case where we can do multi-region placement we really, really should. Would save a lot of grief.

Not totally sure whether this "infrastructure" PR should also be including memory tuning stuff though. Unlike the other bits, there's no dependency between them.